aspnetcore/performance/caching/output.md

```diff
@@ -0,0 +1,155 @@
+
+## Default output caching policy
+
+By default, output caching follows these rules
+
+* Only HTTP 200 responses are cached.
+* Only HTTP GET or HEAD requests are cached.
+* Responses that set cookies aren't cached.
+* Responses to authenticated requests aren't cached.
+
+The following code applies all of the default caching rules to all of an app's endpoints:
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="policies3":::
+
+The following code removes these defaults while applying caching to all of an app's endpoints:
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="policies4":::
+
+
+You can override these defaults.
+
+## Specify the cache key
+
+By default, every part of the URL is included as the key to a cache entry -- that is, the scheme, host, port, path, and query string. However, you might want to explicitly control the cache key. For example, suppose you have an endpoint that returns a unique response for each unique value of the `culture` query string. Variation in other parts of the URL, such as other query strings, shouldn't result in different cache entries. You can specify such rules in a policy, as shown in the following highlighted code:
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="policies2" highlight="7":::
+
+You can then select the `VaryByQuery` policy for an endpoint:
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="selectquery":::
+
+Here are some of the options for controlling the cache key:
+
+* <xref:Microsoft.AspNetCore.OutputCaching.OutputCachePolicyBuilder.SetVaryByQuery%2A> - Specify one or more query string names to act as the cache key.
+* <xref:Microsoft.AspNetCore.OutputCaching.OutputCachePolicyBuilder.SetVaryByHeader%2A> - Specify one or more HTTP headers to act as the cache key.
+* <xref:Microsoft.AspNetCore.OutputCaching.OutputCachePolicyBuilder.VaryByValue%2A>- Specify a value to act as the cache key. The following example uses a value that indicates whether the current server time in seconds is odd or even. A new response is generated only when the number of seconds goes from odd to even or even to odd.
+
+  :::code language="csharp" source="output/samples/7.x/Program.cs" id="varybyvalue":::
+
+For more options, see the <xref:Microsoft.AspNetCore.OutputCaching.OutputCachePolicyBuilder> class.
+
+## Cache revalidation
+
+Cache revalidation means the server can return a `304 Not Modified` HTTP status code instead of the full response body. This status code informs the client that the response to the request is unchanged from what the client previously received.
+
+The following code illustrates the use of an `Etag` header to enable cache revalidation. If the client sends an `If-None-Match` header with the etag value of an earlier response, and the cache entry is fresh, the server returns 304 instead of the full response:
```

Comment:
```suggestion
The following code illustrates the use of an [`Etag`](https://developer.mozilla.org/docs/Web/HTTP/Headers/ETag) header to enable cache revalidation. If the client sends an [`If-None-Match`](https://developer.mozilla.org/docs/Web/HTTP/Headers/If-None-Match) header with the etag value of an earlier response, and the cache entry is fresh, the server returns [304 Not Modified](https://developer.mozilla.org/docs/Web/HTTP/Status/304) instead of the full response:
```